File: _exercises/08_vision/08_vision.py
# %%
import chatlas
import dotenv
from pyhere import here

dotenv.load_dotenv()

# %%
recipe_image = here("data/recipes/images/ClassicBakedZiti.jpg")

# %%
chat = chatlas.ChatOllama(model="gemma3:4b")
chat.chat(
    "Give the food in this image a creative recipe title.",
    chatlas.content_image_file(recipe_image),
)

# %%
chat = chatlas.ChatOllama(model="gemma3:4b")
chat.chat(
    "Write a recipe for the meal in this image.",
    chatlas.content_image_file(recipe_image),
)

File: _exercises/08_vision/08_vision.R
library(ellmer)

recipe_image <- here::here("data/recipes/images/EasyBasicPancakes.jpg")

chat <- chat_ollama(model = "gemma3:4b")
chat$chat(
  "Describe this image in detail",
  content_image_file(recipe_image)
)

chat <- chat_ollama(model = "gemma3:4b")
chat$chat(
  "Write a recipe to make the food in this image",
  content_image_file(recipe_image)
)
